demo/src/app/components/typeahead/overview/sections/customize-options-list/template.html

```diff
@@ -0,0 +1,41 @@
+<p>
+  As said above, the options list can contain any kind of item, and this is handy since selecting an option will commit the actual item to the data model, allowing to store complex objects.
+</p>
+
+<p>
+  But the requirement is to be able to display those options, be it in the list of the popup or inside the input field.
```

Comment:
Two paragraphs too complex, maybe → 

Typeahead might use complex objects as options, so it would need to transform them to HTML to display in the results popup. 

If your typeahead uses strings there is nothing to do, otherwise you need to either provide `[resultFormatter]` function or a `[resultTemplate]`

Reply:
I can simplify the paragraphs. But in your proposition, there are shortcuts which make the statement not exact: 

- transform to HTML: it's either transformed to a string or using a template, both being different from going through an HTML representation
- "there is nothing to do": a but confusing, if they want to customize, they still have to do something.

But these are details, I got the idea, I will simplify. However I am still willing to highlight the fact that the model (`ngModel`) is storing the actual, initial value provided by the developer, not any representation.